File: trunk/src/scala/org/codehaus/testdox/intellij/AbstractTestElement.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.actionSystem.Presentation

abstract class AbstractTestElement extends TestElement {

  val psiElement = NullPsiElement.INSTANCE

  def jumpToPsiElement() = false

  def updatePresentation(presentation: Presentation): Unit = presentation.setEnabled(false)

  def rename(controller: TestDoxController) {}

  def delete(controller: TestDoxController) {}

  def compareTo(o: TestElement): Int = if (this == o) 0 else -1

  override def toString = displayString
}

File: branches/pre-scala-migration/src/java/org/codehaus/testdox/intellij/AbstractTestElement.java
package org.codehaus.testdox.intellij;

import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.psi.PsiElement;

public abstract class AbstractTestElement implements TestElement {

    public PsiElement getPsiElement() {
        return NullPsiElement.INSTANCE;
    }

    public boolean jumpToPsiElement() {
        return false;
    }

    public void updatePresentation(Presentation presentation) {
        presentation.setEnabled(false);
    }

    public void rename(TestDoxController testDoxController) { }

    public void delete(TestDoxController testDoxController) { }

    public int compareTo(Object object) {
        return (equals(object) ? 0 : -1);
    }

    public String toString() {
        return getDisplayString();
    }
}
